tests/postgres_tests/test_json.py

```diff
@@ -299,6 +299,24 @@ def test_regex(self):
     def test_iregex(self):
         self.assertTrue(JSONModel.objects.filter(field__foo__iregex=r'^bAr$').exists())
 
+    def test_deep_lookup_qset_rhs(self):
+        # See ticket #28872
+        Character.objects.create(name='bar')
+
+        query = JSONModel.objects.filter(
+            field__foo__in=Character.objects.values_list('name')
+        )
+
+        self.assertSequenceEqual(query, [self.objs[11]])
+
+    def test_deep_lookup_direct_value_rhs(self):
```

Comment:
This test seems to be passing without the patch (fine to add additional test coverage but we'd rather do it in a separate commit so it's clear which case the patch fixes).

Reply:
Makes sense - will remove this test from the commit.